File: docs-examples/example-config-java/src/test/java/io/micronaut/configuration/openapi/docs/config/PropertiesConfigurationTest.java
package io.micronaut.configuration.openapi.docs.config;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class PropertiesConfigurationTest {

    @Test
    void placeholdersAreExpanded() {
        var info = OpenApiSpec.load("hello-world-v1.1.yml").getInfo();

        assertEquals("Hello World", info.getTitle());
        assertEquals("A nice API", info.getDescription());
        assertEquals("v1.1", info.getVersion());
        assertEquals("Fred", info.getContact().getName());
        assertEquals("Apache 2.0", info.getLicense().getName());
    }
}

File: docs-examples/example-config-kotlin/src/test/kotlin/io/micronaut/configuration/openapi/docs/config/PropertiesConfigurationTest.kt
package io.micronaut.configuration.openapi.docs.config

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class PropertiesConfigurationTest {

    @Test
    fun placeholdersAreExpanded() {
        val info = OpenApiSpec.load("hello-world-v1.1.yml").info

        assertEquals("Hello World", info.title)
        assertEquals("A nice API", info.description)
        assertEquals("v1.1", info.version)
        assertEquals("Fred", info.contact.name)
        assertEquals("Apache 2.0", info.license.name)
    }
}
